src/moveFile.js

```diff
@@ -0,0 +1,99 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+
+function copyFileContentAndMove(source, destination) {
+  try {
+    const fileContent = fs.readFileSync(source, (err) => {
+      if (err) {
+        return err;
+      }
+    });
+
+    fs.writeFileSync(destination, fileContent, (err) => {
+      if (err) {
+        return err;
+      }
+    });
+
+    fs.unlink(source, (err) => {
+      if (err) {
+        return err;
+      }
+    });
+  } catch (error) {
+    throw error;
+  }
+}
+
+function moveFile(command, source, destination) {
+  if (command === 'mv') {
+    const sourcePath = path.dirname(source);
+    const destinationPath = path.dirname(destination);
+    const sourceExt = path.extname(source);
+    const destinationExt = path.extname(destination);
+    const sourceFileName = path.basename(source, path.extname(source));
+    let destinationFileName = '';
+
+    if (destination[destination.length - 1] !== '/'
+      && destinationPath !== '.') {
+      destinationFileName = path.basename(
+        destination,
+        path.extname(destination)
+      );
+    }
+
+    let updatedDestinationPath = destination;
+
+    let destinationCheckResult = false;
+
+    destinationCheckResult = fs.existsSync(destination);
+
+    if (destinationExt !== sourceExt && destinationPath !== '.') {
+      updatedDestinationPath = path.join(
+        destinationPath,
+        destinationFileName + sourceExt
+      );
+    }
+
+    if (!destinationFileName && destinationCheckResult) {
+      updatedDestinationPath = path.join(
+        destination,
+        sourceFileName + sourceExt
+      );
+    }
+
+    switch (true) {
```

Comment:
Switch statement is not the best choice here. Consider using if-else statements.

Reply:
Done